### PR TITLE
feat: [PIE-3668]: Git popover changes, bugfixes, unit tests

### PR DESCRIPTION
**Pull request by depanshu-dhiman**
##### Summary:
1. Git popover change
2. Alignment issue in firefox
3. Reduce save pipeline modal height
4. Reset form errors on save pipeline radio button change - Commit to existing / new
5. Same target branch issue
6. RCC refactor
7. Test cases WIP and snapshot update

##### Jira Links:
https://harness.atlassian.net/browse/PIE-3671
https://harness.atlassian.net/browse/PIE-3883

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/10-common/components/GitRemoteDetails/__tests__/GitRemoteDetails.test.tsx [hunk 1]
```diff
@@ -38,4 +38,55 @@ describe('GitRemoteDetails test', () => {
     expect(getByText('testRepoName')).toBeInTheDocument()
     expect(container).toMatchSnapshot()
   })
+
+  test('readonly', () => {
+    const { container, getByText } = render(
+      <TestWrapper
+        path="/account/:accountId/ci/orgs/:orgIdentifier/projects/:projectIdentifier/pipelines/-1/pipeline-studio/"
```
Comment (chetanvekariya-harness) [inline src/modules/10-common/components/GitRemoteDetails/__tests__/GitRemoteDetails.test.tsx:45]: What is this indentation ? 
Also, could we please store this path in variable and use variable ?

File: src/modules/10-common/components/SaveToGitFormV2/SaveToGitFormV2.tsx [hunk 2]
```diff
@@ -50,6 +51,22 @@ interface ModalConfigureProps {
...
+  isEditing: boolean,
+  getString: (key: keyof StringsMap, vars?: Record<string, any> | undefined) => string
+): SaveToGitFormV2Interface => ({
+  isNewBranch: false,
+  branch: defaultTo(resource.gitDetails?.branch, ''),
+  commitMsg: getString(isEditing ? 'common.gitSync.updateResource' : 'common.gitSync.createResource', {
+    name: resource.name,
+    type: getEntityNameFromType(resource.type)
+  }),
+
```
Comment (chetanvekariya-harness) [inline src/modules/10-common/components/SaveToGitFormV2/SaveToGitFormV2.tsx:65]: NIT - Please remove this empty line. Empty lines in objects is not ideal.

File: src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/__tests__/PipelineCanvas.test.tsx [hunk 3]
```diff
@@ -298,3 +334,75 @@ describe('Existing pipeline', () => {
     expect(queryByText('unsavedChanges')).toBeTruthy()
   })
 })
+
+describe('Git simplication', () => {
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/__tests__/PipelineCanvas.test.tsx:338]: Take these to separate file and dedicate that too new git-sync. You may take a reference from other file (src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/__tests__/PipelineCanvasGitSync.test.tsx) which is dedicated to old git-sync

File: src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/__tests__/PipelineCanvasTestHelper.ts [hunk 4]
```diff
@@ -178,3 +178,34 @@ export const mockPipelineTemplateYaml: UseGetMockDataWithMutateAndRefetch<Respon
...
+      ...data.state,
+      storeMetadata: {
+        connectorRef: 'harness',
+        storeType: 'REMOTE'
+      },
+      ...(pipelineError ? { pipeline: {}, templateError: { data: {}, message: '' } } : {})
+    },
+    updateGitDetails: jest.fn(),
+    updatePipelineStoreMetadata: jest.fn()
+    // updateGitDetails: jest.fn(),
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/__tests__/PipelineCanvasTestHelper.ts:208]: Remove these comments.

File: src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/__tests__/PipelineCanvasGitSimplification.test.tsx [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+jest.mock('@common/hooks', () => ({
+  ...(jest.requireActual('@common/hooks') as any),
+  useMutateAsGet: jest.fn()
+}))
+
+const showError = jest.fn()
+const showSuccess = jest.fn()
+const toasterClear = jest.fn()
+
+jest.mock('@wings-software/uicore', () => ({
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/__tests__/PipelineCanvasGitSimplification.test.tsx:101]: I will remove these kind of junk from old files and I will also remove from this new file. This is not needed at all and it is wrong to do this as well.

File: src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/__tests__/PipelineCanvasGitSimplification.test.tsx [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+          <PipelineCanvas {...props} />
+        </PipelineContext.Provider>
+      </TestWrapper>
+    )
+    expect(openModal).not.toHaveBeenCalled()
+    const editButton = container.querySelector('[data-icon="Edit"]')?.parentElement as HTMLButtonElement
+    act(() => {
+      fireEvent.click(editButton)
+    })
+    expect(openModal).toBeCalledTimes(1)
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/__tests__/PipelineCanvasGitSimplification.test.tsx:175]: After this let's also check that all fields like connectorRef etc are there and they are disabled.

File: src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/__tests__/PipelineCanvasGitSimplification.test.tsx [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+        </PipelineContext.Provider>
+      </TestWrapper>
+    )
+    expect(container.querySelector('.normalInputStyle')).toBeInTheDocument()
+    expect(container.querySelector('[data-icon="git-new-branch"]')).toBeInTheDocument()
+    expect(getByText('pipeline.gitExperience.noEntityFound')).toBeInTheDocument()
+    expect(getByText('pipeline.gitExperience.selectDiffBranch')).toBeInTheDocument()
+    const remoteBranchInput = container.querySelector('input[name="remoteBranch"]') as HTMLInputElement
+    expect(remoteBranchInput).toBeInTheDocument()
+    expect(remoteBranchInput.value).toBe(gitQueryParams.branch)
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/__tests__/PipelineCanvasGitSimplification.test.tsx:154]: We can change branch after this and check if pipeline studio is getting rendered for the correct branch.

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Pull request merged**